docs/GETTING_STARTED.md

```diff
@@ -1,288 +1,247 @@
 ```
 
 Alternately, you can split up stdout (json logs) and stderr (plain text errors):
 
+```shell
+sudo soci-snapshotter-grpc 2> ~/soci-snapshotter-errors 1> ~/soci-snapshotter-logs &
 ```
-sudo ./soci-snapshotter-grpc 2> ~/soci-snapshotter-errors 1> ~/soci-snapshotter-logs &
-```
 
-Once the snapshotter is running we can call the rpull command from the SOCI CLI.
+### Lazily pull image
+
+Once the snapshotter is running we can call the `rpull` command from SOCI CLI.
 This command reads the manfiest from the registry and mounts FUSE filesystems
 for each layer.
 
-In this command the `--soci-index-digest` needs to be the digest of the SOCI
-index manifest.  The final argument is the tag of the image:
-
-**NOTE: Specifying `--soci-index-digest` is optional. If the argument is not provided, the snapshotter will use the [Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers) to fetch a list of available indices.**
-```
-sudo ./soci image rpull --plain-http --soci-index-digest sha256:f5f2a8558d0036c0a316638c5575607c01d1fa1588dbe56c6a5a7253e30ce107 localhost:5000/rabbitmq:latest
-```
+> The optional flag `--soci-index-digest` needs to be the digest of the SOCI index manifest.
+> If not provided, the snapshotter will use the [Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers) to fetch a list of available indices.
 
-After running this command you will see a minimal output.  If you see all of the
-layers being fetched now, then SOCI has not worked.  It should look something
-like:
+```shell
+sudo soci image rpull --user AWS:$ECR_PASS --soci-index-digest sha256:f5f2a8558d0036c0a316638c5575607c01d1fa1588dbe56c6a5a7253e30ce107 $REGISTRY/rabbitmq:latest
 
-```
+# output
 fetching sha256:7e2a3a93... application/vnd.docker.distribution.manifest.v2+json
 fetching sha256:31b721ac... application/vnd.docker.container.image.v1+json
 ```
 
-Now we should also be able to see the mounts that were created for the FUSE
-filesystems.  There should be one mount for each layer.  To see this use the
-`mount` command.  Within the output you should see lines like:
+After running this command you will see a minimal output as the example, because
+with lazy pulling, not all layers are fetched/pulled during the `pull` step. If
+you see all layers being fetched, then SOCI is not working. (TODO: add possible solution)
 
-```
+Now let's check the mounts for the FUSE filesystems. There should be one mount per layer.
+
+```shell
+mount | grep fuse
+
+# output
 /home/ec2-user/soci-snapshotter/out/soci on /var/lib/soci-snapshotter-grpc/snapshotter/snapshots/1/fs type fuse.rawBridge (rw,relatime,user_id=0,group_id=0,allow_other)
 fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
 /home/ec2-user/soci-snapshotter/out/soci on /var/lib/soci-snapshotter-grpc/snapshotter/snapshots/2/fs type fuse.rawBridge (rw,relatime,user_id=0,group_id=0,allow_other)
-.....
+...
```

Comment:
We only expect 3 mounts. Can we show all 3?

Reply:
updated